stance/motion/squat.py

```diff
@@ -0,0 +1,103 @@
+from .motion import Motion
+import numpy as np
+
+
+class Squat(Motion):
+    def __init__(self, front_view_points, profile_view_points):
+        self.template_skeletons = self.create_template_skeletons(front_view_points, profile_view_points)
+        self.benchmark_zones = [self._0, self._1, self._2, self._3, self._4]
+        self.similarity_scorers = [self._0_similarity, self._1_similarity,
+                                   self._2_similarity, self._3_similarity, self._4_similarity]
+        self.scorers = [self._0_score, self._1_score,
+                        self._2_score, self._3_score, self._4_score]
+
+    # --------------- IMPLEMENTED ABSTRACT METHODS ------------------
+
+    def find_benchmark_zones_of_user(self, user_skeletons):
+        """
+        SEE MOTION DOCSTRING
+        """
+        benchmark_zone_indices = []
+        for similarity_scorer in self.similarity_scorers:
+            score_array = np.array([])
+            for user_skeleton in user_skeletons:
+                np.append(score_array, similarity_scorer(user_skeleton))
+            index_array = np.argmin(score_array)
+            benchmark_zone_indices.append(index_array[0])
+        return benchmark_zone_indices
+
+    def create_template_skeletons(self, front_view_points, profile_view_points):
+        return [zone(front_view_points, profile_view_points) for zone in self.benchmark_zones]
+
+    def score(self, user_skeletons):
+        score = 0
+        for i in range(len(self.scorers)):
+            score += self.scorers[i](user_skeletons[i])
+        return score
+
+    # --------------- BENCHMARK ZONES ------------------
+
+    def _0(self, front_view_points, profile_view_points):
```

Comment:
I hate this convention for benchmark zones. We need to think of something better.  Probably later though